### PR TITLE
chore: enforce import ordering with eslint

### PR DESCRIPTION
**Pull request by nolanlawson**
## Details

Uses [eslint-plugin-import](https://github.com/benmosher/eslint-plugin-import) to ensure that imports occur in the following order:

- builtins
- external
- internal

This is [borrowed](https://github.com/airbnb/javascript/blob/1eadb93/packages/eslint-config-airbnb-base/rules/imports.js#L146-L149) from `eslint-config-airbnb-base`.

I saw this come up in a few code reviews, so I figured we should formalize this. `eslint --fix` can fix these ordering issues automatically.

## Does this PR introduce breaking changes?

* ✅ `No, it does not introduce breaking changes.`

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -90,7 +91,11 @@
         "template-curly-spacing": "error",
         "yoda": "error",
 
-        "lwc-internal/no-invalid-todo": "error"
+        "lwc-internal/no-invalid-todo": "error",
+        "import/order": [
+            "error", { "groups": [[
+                "builtin", "external", "internal"
+            ]] }]
```
Comment (pmdartus) [inline .eslintrc:98]: Nitpicking on the formatting.

```suggestion
        "import/order": [
            "error", 
            { "groups": [["builtin", "external", "internal"]] }
        ]
```

**Review by pmdartus — APPROVED** [review]

**Review by ekashida — APPROVED** [review]
Nice!

**Comment by abdulsattar** [comment]
Can we look into using prettier to do this? It will sort the imports automatically rather than us manually having to sort them.

**Comment by pmdartus** [comment]
Prettier won't touch the order of the imports. The new rule that @nolanlawson is introducing is fixable so you can run `yarn lint --fix` from the root of the fix to get the order right automatically.

**Comment by abdulsattar** [comment]
Looks like it is possible: https://www.npmjs.com/package/prettier-plugin-import-sort, https://github.com/trivago/prettier-plugin-sort-imports.

Maybe we can explore this in another PR.

**Review by abdulsattar — APPROVED** [review]

**Comment by pmdartus** [comment]
Ohh I see as a prettier plugin.

**Comment by nolanlawson** [comment]
I don't have a strong opinion about whether this should be in eslint or prettier.

**Comment by nolanlawson** [comment]
Merging for now, we can look into Prettier later.

**Pull request merged**